File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        nums = [1,2,2,3,3,4,4,4,4];
        const newSet = new Set(nums);
        let count = {};
        let freq = Array.from({length: nums.length + 1}, () => []);
        
        for(const n of nums) {
            count[n] = (count[n] || 0) + 1;
        }

        for(const n in count) {
            freq[count[n]].push(parseInt(n))
        }
        console.log(freq);
        const res = [];
        for (let i = freq.length - 1; i > 0; i--) {
            for(const n of freq[i]) {
                res.push(n);
                if(res.length == k) return res;
            }
        }

    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        seen = set()
        l = 0
        res = 0
        if len(s) == 1:
            return 1
        if len(s) == 2:
            if s[0] == s[1]:
                return 1
            else:
                return 2
        for r in range(1, len(s) - 1):
            if s[r] not in seen:
                res += 1
                seen.add(s[r])
            else:
                l = r + 1
        return res
        